ansible/content-servers.yml

```diff
@@ -0,0 +1,197 @@
+- hosts: bananas_fileservers
+  gather_facts: true
+  become: true
+
+  vars:
+    # Our vars, for easy reuse
+    cache_timeout: "1y"  # Basically just as long as nginx runs for
+    bananas_server: bananas.openttd.org
+    bananas_cdn_server: bananas.cdn.openttd.org
+    bananas_server_secret: !vault |
+      $ANSIBLE_VAULT;1.1;AES256
+      36643134646634303835373564336465653938376530363333353830306134336461333730633662
+      3139636438373063306331386339366331343231383766390a306634346539353361336431393662
+      66313332353438623662653138333239623330613361613837346162373261626266646631653835
+      3162343533373631360a376136363233313937333032356137353934373536616164373166633065
+      30353033393436663230363436356265363739663266363034326263356163326264
+
+    nginx_shared_config: |
+      location = /healthz {
+        access_log off;
+        return 200;
+      }
+      # Rewrite the first to the second:
+      #   base-graphics/12345678/12345678901234567890123456789012/filename.tar.gz
+      #   base-graphics/12345678/12345678901234567890123456789012.tar.gz
+      # This allows the OpenTTD client to know the name to use for the file,
+      #   while the S3 only knows the md5sum based name.
+      location / {
+        rewrite "^/([a-z-]+)/([a-f0-9]{8})/([a-f0-9]{32})/[a-zA-Z0-9-_.]+\.tar\.gz$" $1/$2/$3.tar.gz break;
+        return 404;
+
+        proxy_pass https://{{ bananas_cdn_server }}/;
+        proxy_hide_header Strict-Transport-Security;
+
+        proxy_cache nginx-cache;
+        proxy_cache_valid 200 {{ cache_timeout }};
+        proxy_cache_use_stale error timeout updating http_500 http_502 http_503 http_504;
+        proxy_cache_revalidate on;
+        proxy_cache_lock on;
+        proxy_ignore_headers Set-Cookie;
+        add_header X-Cache-Status $upstream_cache_status;
+      }
+
+    # Role config vars
+    certbot_admin_email: info@openttd.org
+    certbot_create_if_missing: true
+    certbot_create_standalone_stop_services: []
+    certbot_certs:
+      - domains:
+          - "{{ inventory_hostname }}"
+    nginx_extra_http_options: |
+      proxy_cache_path /tmp/nginx-cache/ levels=1:2 keys_zone=nginx-cache:16m max_size=1g inactive={{ cache_timeout }} use_temp_path=off;
+    nginx_remove_default_vhost: true
+    nginx_vhosts:
+      - listen: "80"
+        server_name: "{{ inventory_hostname }}"
+        filename: "{{ inventory_hostname }}.80.conf"
+        extra_parameters: |
+          {{ nginx_shared_config }}
+
+      - listen: "443 ssl http2"
+        server_name: "{{ inventory_hostname }}"
+        filename: "{{ inventory_hostname }}.443.conf"
+        # Config partially taken from https://ssl-config.mozilla.org/#server=nginx&version=1.14.2&config=intermediate&openssl=1.1.1d&guideline=5.6
```

Comment:
The word "partially" always scares me, as it gives me no indication what you didn't do, and why that would be. I don't mind this comment, but maybe add what you didn't add and why?